frontend/src/scenes/feature-flags/featureFlagLogic.ts

```diff
@@ -191,7 +195,18 @@ export const featureFlagLogic = kea<featureFlagLogicType>([
                 },
             }),
             submit: (featureFlag) => {
-                actions.saveFeatureFlag(featureFlag)
+                const newPayload = {}
+                featureFlag.filters.multivariate?.variants.forEach(({ key }, index) => {
+                    const payload = featureFlag.filters.payloads[index]
+                    newPayload[key] = payload
+                })
```

Comment:
you could put this data manipulation into the loader action `saveFeatureFlag` instead and keep this submit clean